flux-desugar/src/desugar.rs

```diff
@@ -1058,6 +1078,19 @@ impl Binder {
         }
     }
 
+    fn from_base(
+        name_gen: &IndexGen<fhir::Name>,
+        map: &fhir::Map,
+        base: &surface::BaseTy<Res>,
+    ) -> Binder {
+        match base {
+            surface::BaseTy::Path(path) => Binder::from_res(name_gen, map, path.ident),
+            surface::BaseTy::Array(_, _) | surface::BaseTy::Slice(_) => {
+                Binder::from_res(name_gen, map, Res::Uint(surface::UintTy::Usize))
```

Comment:
I would inline this call as
```suggestion
                Binder::Single(name_gen.fresh(), fhir::Sort::Int)
```

Reply:
done!